docs/conditions.md

```diff
@@ -32,11 +33,34 @@ following fields:
 ### Check
 
 The `check` field is required. You define a single check to define the body of a `Condition`. The 
-check must specify a container image that adheres to the [container contract](./container-contract.md). The container image 
-runs till completion. The container must exit successfully i.e. with an exit code 0 for the 
-condition evaluation to be successful. All other exit codes are considered to be a condition check
+check must specify a container image that adheres to the [container contract](./container-contract.md). 
+The container image runs till completion. The container must exit successfully i.e. with an exit code 0 
+for the condition evaluation to be successful. All other exit codes are considered to be a condition check
 failure.
 
+### Parameters
+
+A Condition can declare parameters that must be supplied to the it during a PipelineRun. Sub-fields
+within the check field can access the parameter values using the templating syntax:
+
+```yaml
+spec:
+  parameters:
+    - name: image
+      default: ubuntu
+  check:
+    image: ${params.image}
+```
+
+Parameters name are limited to alpha-numeric characters, `-` and `_` and can
+only start with alpha characters and `_`. For example, `fooIs-Bar_` is a valid
+parameter name, `barIsBa$` or `0banana` are not.
+ 
+Each declared parameter has a type field, assumed to be string if not provided by the user. 
+The other possible type is array — useful, for instance, when a dynamic number of compilation 
```

Comment:
> compilation flags

Would a condition container typically be responsible for compiling something? If not, is there another example that might be more relevant to condition containers?  Maybe something like "checking a pushed branch name doesn't match any of multiple protected branch names" where the protected branch names are passed in as an array param?

Reply:
great idea! and changed!